tests/test_flask_util.py

```diff
@@ -0,0 +1,328 @@
+
+    def test_no_configuration(self):
+        self.assertRaises(
+            ValueError,
+            FlaskOAuth2,
+            flask.Flask(__name__))
+
+    def test_create_flow(self):
+        with self.app.test_request_context():
+            flow = self.oauth2._make_flow()
+            self.assertTrue('google_oauth2_state' in flask.session)
+            self.assertEqual(
+                flask.session['google_oauth2_state'], flow.params['state'])
+            self.assertEqual(flow.client_id, self.oauth2.client_id)
+            self.assertEqual(flow.client_secret, self.oauth2.client_secret)
+            self.assertTrue('http' in flow.redirect_uri)
+            self.assertTrue('oauth2callback' in flow.redirect_uri)
+
+            flow = self.oauth2._make_flow(extra_arg='test')
+            self.assertEqual(flow.params['extra_arg'], 'test')
+
+        # Test extra args specified in the constructor.
+        app = flask.Flask(__name__)
+        app.config['SECRET_KEY'] = 'notasecert'
+        oauth2 = FlaskOAuth2(
+            app, client_id='client_id', client_secret='secret',
+            extra_arg='test')
+
+        with app.test_request_context():
+            flow = oauth2._make_flow()
+            self.assertEqual(flow.params['extra_arg'], 'test')
+
+    def test_authorize_view(self):
+        with self.app.test_client() as c:
+            rv = c.get('/oauth2authorize')
+            location = rv.headers['Location']
+            self.assertTrue(GOOGLE_AUTH_URI in location)
+            self.assertTrue(self.oauth2.client_id in location)
+            self.assertFalse(self.oauth2.client_secret in location)
+            self.assertTrue(flask.session['google_oauth2_state'] in location)
+            self.assertEqual(flask.session['google_oauth2_return'], '/')
+
+        with self.app.test_client() as c:
+            rv = c.get('/oauth2authorize?return_url=/test')
+            self.assertEqual(flask.session['google_oauth2_return'], '/test')
+
+        with self.app.test_client() as c:
+            rv = c.get('/oauth2authorize?extra_param=test')
+            location = rv.headers['Location']
+            self.assertTrue('extra_param=test' in location)
+
+    def test_logout_view(self):
+        self.oauth2.storage = mock.Mock()
+        self.oauth2.logout_callback = mock.Mock()
+
+        with self.app.test_client() as c:
+            c.get('/oauth2logout')
+            self.oauth2.storage.delete.assert_called()
+            self.oauth2.logout_callback.assert_called()
+
+    def test_callback_view(self):
+        self.oauth2.storage = mock.Mock()
+        self.oauth2.authorize_callback = mock.Mock()
+
+        with self.app.test_client() as c:
+            with Http2Mock() as http:
+                with c.session_transaction() as session:
+                    session['google_oauth2_state'] = 'statez'
+
+                rv = c.get('/oauth2callback?state=statez&code=codez')
+
+                self.assertEqual(rv.status_code, 302)
```

Comment:
httplib.FOUND rather than 302, please. Please use symbolic constants rather than literal values for all HTTP status codes?